pkg/segment/writer/metrics/metricssegment.go

```diff
@@ -524,15 +524,11 @@ func ExtractOTSDBPayload(rawJson []byte, tags *TagsHolder) ([]byte, float64, uin
 		case bytes.Equal(key, otsdb_mname), bytes.Equal(key, metric_name_key):
 			switch valueType {
 			case jp.String:
-				temp, err := jp.ParseString(value)
+				_, err := jp.ParseString(value)
 				if err != nil {
 					log.Errorf("failed to extract tags %+v", err)
 				}
-				if temp != "target_info" {
-					mName = value
-				} else {
-					return nil
-				}
+				mName = value
```

Comment:
this is wrong, if we have an error to parsestring, then whats the point to assign the value to mname, you should just return err and not continue processing this metric